Make CUDA log New Epoch and add --keep-working (-KW) option

SnowLeopard71 opened the pull request:
- CLMiner.cpp logs new epoch, this is similar to CUDAMiner.cpp
- new command line option:
> -KW, --keep-working Do not stop working if server connection lost. Avoids DAG regeneration but continues using power while disconnected.

smurfy commented:
Thanks for the contribution, but doesn't this make more sense in farm::setWork ?
For both Miners?

SnowLeopard71 commented:
Apologies for messing with git, but I think I cleaned it up okay -- squashed into 1 commit.
I also added new command-line option: 
`-KW, --keep-working Do not stop working if server connection lost. Avoids DAG regeneration but continues using power while disconnected.`
Not sure I defined `g_keepworking` properly.
